### PR TITLE
Implement file normalizer and download

### PR DESCRIPTION
**Pull request by sepal**
This PR adds normalizers for managed file and image fields, which also copy the referenced files to the temporary export directory, so that they are zipped with the rest of the export.

To provide the directory where the files should be copied to, I had to rewrite the existing functions. `hook_gdpr_export_user_export()` implementations now accept a `$context` parameter instead of the export directory, so that more options can be passed to those functions. They also can be passed to the serializer and the default context from the `gdpr_export_user_export()` function can be altered.

### CODE REVIEW

**Comment by sepal** [comment]
@klausi can you review this PR? After merging the module should be ready for a first alpha or beta release on d.o.

File: gdpr_export.api.php [hunk 1]
```diff
@@ -120,6 +119,19 @@ function hook_gdpr_export_user_export_format_alter(&$format) {
...
+ * Alter the contexts passed to the hook_gdpr_export_user_export()
+ * implementations and serializations. Encoders and normalizers will therefore
+ * be able to access those options.
+ *
+ * @param array $context
+ *   An array of contexts to alter. The default context gdpr_export_dir contains
+ *   the temp directory to which files copied or exported for zipping.
+ */
+function hook_gdpr_export_user_export_context_alter(&$context) {
+  $context['some_context'] = 'context_value';
```
Comment (klausi) [inline gdpr_export.api.php]: we should use the gdpr_export_dir as example here.

File: gdpr_export.module [hunk 2]
```diff
@@ -80,20 +83,35 @@ function gdpr_export_user_export($account) {
...
+  $real_dir = drupal_realpath($directory);
+
+  // Get a list of all files in the export dir.
+  $files = new RecursiveIteratorIterator(
+    new RecursiveDirectoryIterator($real_dir),
+    RecursiveIteratorIterator::LEAVES_ONLY
+  );
+
+  foreach ($files as $name => $file)
+  {
```
Comment (klausi) [inline gdpr_export.module]: opening braces should be on the same line

File: gdpr_export.module [hunk 3]
```diff
@@ -80,20 +83,35 @@ function gdpr_export_user_export($account) {
...
+    // Skip directories (they will be added automatically)
+    if (!$file->isDir())
+    {
```
Comment (klausi) [inline gdpr_export.module]: same here.

Install Coder? ;-)

File: gdpr_export.module [hunk 4]
```diff
@@ -64,10 +64,13 @@ function gdpr_export_user_export($account) {
   // supported.
   drupal_alter('gdpr_export_user_export_format', $format);
 
+  $context = ['gdpr_export_dir' => $directory];
```
Comment (klausi) [inline gdpr_export.module:67]: I miss the Github syntax hghlighting here. Can you add .gitattributes as in https://github.com/jobiqo/recruiter_internal/blob/7.x-4.x/.gitattributes ?

**Review by klausi — APPROVED** [review]
Looks good, just minor stuff.

File: gdpr_export.module (hunk 4 above)
Comment (sepal) [reply gdpr_export.module:67]: Thats cool, didn't this was possbile!

**Pull request merged**